components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -246,51 +250,41 @@ - (void)setSnackbarView:(MDCSnackbarMessageView *)snackbarView {
+    CGFloat leftMargin = sideMargin;
+    CGFloat rightMargin = sideMargin;
 
     UIView *container = self.containingView;
 
     if (snackbarView) {
       [container addSubview:snackbarView];
 
-      // Pin the snackbar to the bottom of the screen.
+      // Pin the Snackbar to the bottom of the screen.
       [snackbarView setTranslatesAutoresizingMaskIntoConstraints:NO];
 
-      if (fullWidth) {
-        CGFloat leftMargin = sideMargin;
-        CGFloat rightMargin = sideMargin;
-
-#if defined(__IPHONE_11_0) && (__IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_11_0)
-        if (@available(iOS 11.0, *)) {
-          leftMargin += self.safeAreaInsets.left;
-          rightMargin += self.safeAreaInsets.right;
-        }
-#endif
-
-        [container addConstraint:[NSLayoutConstraint constraintWithItem:snackbarView
-                                                              attribute:NSLayoutAttributeLeading
-                                                              relatedBy:NSLayoutRelationEqual
-                                                                 toItem:container
-                                                              attribute:NSLayoutAttributeLeading
-                                                             multiplier:1.0
-                                                               constant:leftMargin]];
-
-        [container addConstraint:[NSLayoutConstraint constraintWithItem:snackbarView
-                                                              attribute:NSLayoutAttributeTrailing
-                                                              relatedBy:NSLayoutRelationEqual
-                                                                 toItem:container
-                                                              attribute:NSLayoutAttributeTrailing
-                                                             multiplier:1.0
-                                                               constant:-1 * rightMargin]];
-      } else {
-
-        [container addConstraint:[NSLayoutConstraint constraintWithItem:snackbarView
-                                                              attribute:NSLayoutAttributeCenterX
-                                                              relatedBy:NSLayoutRelationEqual
-                                                                 toItem:container
-                                                              attribute:NSLayoutAttributeCenterX
-                                                             multiplier:1.0
-                                                               constant:0]];
+      BOOL isRegularWidth =
+          self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassRegular;
+      BOOL isRegularHeight =
+          self.traitCollection.verticalSizeClass == UIUserInterfaceSizeClassRegular;
+      if (isRegularWidth && isRegularHeight) {
+        snackbarView.centerConstraint =
```

Comment:
Style/nit: It feels a little weird to have an externally-available constraint property on the message view object binding it to the presenting container. Would it work to instead track it in the container?

Reply:
Sure, I'll do that